homeassistant/components/mill/sensor.py

```diff
@@ -45,9 +48,10 @@ def __init__(self, heater, mill_data_connection, sensor_type):
             "model": f"generation {1 if heater.is_gen1 else 2}",
         }
 
-    async def async_update(self):
-        """Retrieve latest state."""
-        heater = await self._conn.update_device(self._id)
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        heater = self.coordinator.data[self._id]
         self._attr_available = heater.available
```

Comment:
Break this out into a `_update_attr` too just like in climate and call it in the constructor.